pkg/sharding/listwatch.go

```diff
@@ -0,0 +1,88 @@
+package sharding
+
+import (
+	"crypto/md5"
+
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/tools/cache"
+)
+
+type shardedListWatch struct {
+	sharding *sharding
+	lw       cache.ListerWatcher
+}
+
+func NewShardedListWatch(shard, totalShards uint64, lw cache.ListerWatcher) cache.ListerWatcher {
+	// This is an "optimization" as this configuration means no sharding is to
+	// be performed.
+	if shard == 0 && totalShards == 1 {
```

Comment:
I guess if sharding is not set in `main.go`, these two would default to zero. Don't we want to just do `return lw` in that case? Hope I am not missing something.

Reply:
shard defaults to 0 and total shards defaults to 1